ulib/FStar.Bytes.fsti

```diff
@@ -101,8 +101,22 @@ val set_byte:
 
 unfold let op_String_Assignment = set_byte
 
+val reveal_set_byte:
+    b:bytes
+  -> pos:u32 {U32.v pos < length b}
+  -> x:byte
+  -> Lemma
+    (reveal (set_byte b pos x) == Seq.upd (reveal b) (U32.v pos) x)
+    [SMTPat (reveal (set_byte b pos x))]
+
 unfold let index (b:bytes) (i:nat{i < length b}) = get b (U32.uint_to_t i)
 
+val index_reveal:
+    b: bytes
+  -> (i: nat { i < length b })
+  -> Lemma
+    (S.index (reveal b) i == index b i)
+
```

Comment:
I think it's better to write this lemma in terms of `get` rather than `index`: that latter is unfolded to `get` anyway, so it's better to have a more general lemma about the primitive form rather than the specialized lemma about the derived form.

Reply:
You're right. Let me change that now.

Reply:
Then, I have a choice. Should I rather write:
```
val index_reveal:
   b: bytes
-> i: nat { i < length b }
-> Lemma
   (S.index (reveal b) i == get b (U32.uint_to_t i))
```
or:
```
val get_spec:
    b: bytes
 -> pos: u32 { U32.v pos < length b } 
 -> Lemma
    (get b pos == S.index (reveal b) (U32.v pos))

```
(one is a consequence of the other, but my question is more like: where should the conversion between uint32 and nat happen?)

I refrained from enriching the specification of `get` with an `ensures`, fearing that it could break existing proofs.

Reply:
I prefer the style of `get_spec`. This module supports `nat` operations only for legacy code in mitls which is using nats. I would prefer to deprecate the nat operations.

Also, I think you should add it as a post-condition of `get`, to keep with the style of the other functions in this module. (see append, slice, sub etc.)

I don't think we should go for a non-uniform style in this module already for fear of breaking existing proofs ... it's not yet used much in proofs, aside from your code.